src/Mod/TechDraw/TechDrawTools/Command3DLengthDimension.py

```diff
@@ -0,0 +1,125 @@
+        """Run the following code when the command is activated (button press)."""
+        print("3D length dimension activated")
+        if  self.selectionTest():
+            (edges,vertexes) = self.selectionTest()
+            view = Gui.Selection.getSelection()[0]
+            StartPt, EndPt = edges[1].Vertexes[0].Point, edges[1].Vertexes[1].Point
+            extLineVec = EndPt.sub(StartPt)
+            StartPt, EndPt = edges[0].Vertexes[0].Point, edges[0].Vertexes[1].Point
+            dimLineVec = EndPt.sub(StartPt)
+            xAxis = App.Vector(1,0,0)
+            extAngle = degrees(extLineVec.getAngle(xAxis))
+            lineAngle = degrees(dimLineVec.getAngle(xAxis))
+            if extLineVec.y < 0.0:
+                extAngle = 180-extAngle
+            if dimLineVec.y < 0.0:
+               lineAngle = 180-lineAngle
+            if abs(extAngle-lineAngle)>0.1:
+                distanceDim=TechDraw.makeDistanceDim(view,'Distance',vertexes[0].Point,vertexes[1].Point)
+                distanceDim.AngleOverride = True
+                distanceDim.LineAngle = lineAngle
+                distanceDim.ExtensionAngle = extAngle
+                distanceDim.X = (vertexes[0].Point.x+vertexes[1].Point.x)/2
+                distanceDim.Y = (vertexes[0].Point.y+vertexes[1].Point.y)/2
+                distanceDim.recompute()
+                view.requestPaint()
+            Gui.Selection.clearSelection()
+
+    def IsActive(self):
+        """Return True when the command should be active or False when it should be disabled (greyed)."""
+        if App.ActiveDocument:
+            return TechDrawTools.TDToolsUtil.havePage() and TechDrawTools.TDToolsUtil.haveView()
+        else:
+            return False
+
+    def selectionTest(self):
+        '''test correct selection'''
+        if Gui.Selection.getSelection():
+            view = Gui.Selection.getSelection()[0]
+            if Gui.Selection.getSelectionEx():
+                objectList = Gui.Selection.getSelectionEx()[0].SubElementNames
+                if len(objectList)>=2:
+                    edges = []
+                    vertexes = []
+                    for objectString in objectList:
+                        if objectString[0:4] == 'Edge':
+                            edges.append(view.getEdgeBySelection(objectString))
+                        elif objectString[0:6] == 'Vertex':
+                            vertexes.append(view.getVertexBySelection(objectString))
+                    if len(edges) >= 2:
+                        if len(vertexes)<2:
+                            vertexes = []
+                            vertexes.append(edges[0].Vertexes[0])
+                            vertexes.append(edges[0].Vertexes[1])
+                        return(edges,vertexes)
+                    else:
+                        return False
+                else:
+                    return False
+            else:
+                return False
+        else:
+            return False
```

Comment:
```suggestion
        if not Gui.Selection.getSelection():
            return False
        view = Gui.Selection.getSelection()[0]

        if not Gui.Selection.getSelectionEx():
            return False
        objectList = Gui.Selection.getSelectionEx()[0].SubElementNames

        if not len(objectList)>=2:
            return False

        edges = []
        vertexes = []
        for objectString in objectList:
            if objectString[0:4] == 'Edge':
                edges.append(view.getEdgeBySelection(objectString))
            elif objectString[0:6] == 'Vertex':
                vertexes.append(view.getVertexBySelection(objectString))
        if not len(edges) >= 2:

        if len(vertexes)<2:
            vertexes = []
            vertexes.append(edges[0].Vertexes[0])
            vertexes.append(edges[0].Vertexes[1])
        return(edges,vertexes)
```

bruh, do you need me to come back and preach